client.go

```diff
@@ -397,6 +365,62 @@ func (c *Client) do(method, urlpath string, doOptions doOptions) (*http.Response
 	return nil, netutil.ErrAllFailed(addresses)
 }
 
+var errTryNextEndpoint = errors.New("try next endpoint")
```

Comment:
Better to declare errors at the top of file?

Reply:
done.